### PR TITLE
Feature/690 notifications email identified user endpoint

### PR DESCRIPTION
**Pull request by eskebab**
## Related Issue(s)
- #690 

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

## Summary by CodeRabbit

* **New Features**
  * POST endpoint to look up contact points (email and mobile) for self‑identified users via external identities; returns decoded email and associated external identity and handles empty input.
  * Detects and extracts ID‑Porten emails (including URI‑encoded values) from external identifiers.
  * New request/response models for submitting external identities and returning contact points.

* **Chores**
  * Added URN/encoding support and updated project dependencies.

* **Tests**
  * Unit tests covering URN formats, email decoding and edge cases.

### CODE REVIEW

File: src/Altinn.Profile/Controllers/SiContactDetailsLookupCriteria.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+/// Represents the criteria used to look up contact details for self-identified users by their email identifiers.
+/// </summary>
+public class SiContactDetailsLookupCriteria
+{
+    /// <summary>
+    /// A list of email identifiers for which to retrieve contact points for self-identified users.
+    /// </summary>
+    [JsonRequired]
+    [Required]
+    public List<string> EmailIdentifiers { get; set; } = [];
```
Comment (sigridbra) [inline src/Altinn.Profile/Controllers/SiContactDetailsLookupCriteria.cs]: I think the most "future-proof" name here will be `ExternalIdentities` to match the name from the user profile

File: src/Altinn.Profile.Core/User.ContactPoints/SiUserContactPoints.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿﻿namespace Altinn.Profile.Core.User.ContactPoints;
+
+/// <summary>
+/// Represents a set of contact points for a self-identified user, including mobile number and email address.
+/// </summary>
+public class SiUserContactPoints
```
[the remark this answers is not in the document]
Comment (SandGrainOne) [reply src/Altinn.Profile.Core/User.ContactPoints/SiUserContactPoints.cs]: Yeah, let's avoid the need to split the URN to find a match.

[the remark this answers is not in the document]
Comment (eskebab) [reply src/Altinn.Profile.Core/User.ContactPoints/SiUserContactPoints.cs]: I have added externalidentity in the response with the encoded urn string

File: src/Altinn.Profile.Core/User.ContactPoints/SiUserContactPoints.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.InteropServices;
```
[the remark this answers is not in the document]
Comment (eskebab) [reply src/Altinn.Profile.Core/User.ContactPoints/SiUserContactPoints.cs]: all good now?
